src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1242,6 +1242,16 @@ public virtual MethodSymbol TupleUnderlyingMethod
             }
         }
 
+        /// <summary>
+        /// Returns a suitable location for this MethodSymbol for a diagnostic to be placed at. This
+        /// is generally just the first entry in <see cref="Symbol.Locations"/>, but it may be a 
+        /// smaller location if the full location is large.  For example, Locations[0] on  lambda 
+        /// symbols covers the entire syntax, which is inconvenient but remains for compatibility.
+        /// For better diagnostics quality, <see cref="DiagnosticLocation"/> will point to the 
+        /// "delegate" or the "=>".
+        /// </summary>
+        internal virtual Location DiagnosticLocation => Locations[0];
```

Comment:
The one caller that was replaced assumed `Locations` was not empty. But what about callers of this new property (or at least, new to this base class)? How would a caller know it is safe to call `DiagnosticLocation`?